### PR TITLE
Isy 722 refactoring wizard

### PR DESCRIPTION
**Pull request by msg-ptavoul**

### CODE REVIEW

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.ts [hunk 1]
```diff
@@ -134,58 +135,70 @@ export class WizardComponent implements OnInit, AfterContentInit {
...
    */
-  move(next: boolean): void {
-    this.stepper.move(next);
-    this.stepperIndexChange.emit(this.stepper.index);
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.isVisible && this.isVisible) {
+      if (changes.isVisible.currentValue === false) {
+        this.resetWizard();
+      }
+    }
```
Comment (msg-tbrtz) [inline projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.ts:170]: A little hard to read, can't this be simplified?

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts [hunk 2]
```diff
@@ -278,118 +275,112 @@ describe('Integration Tests: WizardComponent with Mock Parent', () => {
...
 
-  it('should not have saved on init', () => {
+  it('should not be saved on init', () => {
     expectIsSaved(false);
   });
 
-  it('should not have the next step available on init', () => {
+  it('should not have an available next step on init', () => {
     expectNextStepIsAllowed(false);
   });
```
Comment (msg-tbrtz) [inline projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:292]: sounds like "should never have an available step", which is wrong. 
also, it is set from outside component, rendering the test useless

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts [hunk 3]
```diff
@@ -419,43 +408,39 @@ describe('Integration Tests: WizardComponent with Mock Parent', () => {
...
-  });
-
   it('should use the correct save button label', () => {
-    expectStepperMovedUntilEnd();
+    expectWizardMovedUntilEnd();
     const saveButton = getNativeElementAsHTMLElement(saveButtonDeclaration);
     expect(saveButton.innerHTML).toContain(wizard.labelSaveButton);
   });
 
   it('should have some classes on the save button', () => {
```
Comment (msg-tbrtz) [inline projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:423]: what is this actually supposed to test and why? sounds like implementation detail

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts [hunk 4]
```diff
@@ -419,43 +408,39 @@ describe('Integration Tests: WizardComponent with Mock Parent', () => {
...
   it('should have a visible save button on last step', () => {
-    expectStepperMovedUntilEnd();
-
+    expectWizardMovedUntilEnd();
     const saveButton = getNativeElementAsHTMLElement(saveButtonDeclaration);
     expect(saveButton).not.toBeNull();
   });
 
-  it('should have disabled save button on last step', () => {
+  it('should have a save button be disabled on last step', () => {
```
Comment (msg-tbrtz) [inline projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:438]: last step not seen in test
why does this test work anyway?

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts [hunk 5]
```diff
@@ -419,43 +408,39 @@ describe('Integration Tests: WizardComponent with Mock Parent', () => {
...
     expectSaveButtonIsAvailable();
     expect(isElementDisabled(saveButtonDeclaration)).toBeTrue();
   });
 
-  it('should have functional save button', () => {
+  it('should have a functional save button', () => {
     const param = true;
```
Comment (msg-tbrtz) [inline projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:444]: this test seems a bit confusing.

you expect a disabled save button by default, then set the next step to available (hardcoded) and now expect save button to be enabled? why so? why would it be even there if you later test it to be null?

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts [hunk 6]
```diff
@@ -488,138 +476,85 @@ describe('Integration Tests: WizardComponent with Mock Parent', () => {
```
Comment (msg-tbrtz) [inline projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:476]: it the line above again not an implementation detail?

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts [hunk 7]
```diff
@@ -488,138 +476,85 @@ describe('Integration Tests: WizardComponent with Mock Parent', () => {
...
-    expect(isCloseButtonDisabled).toBeFalse();
-
+    moveToLastStep();
     pressSaveButton();
     expectIsClosable(true);
+  });
 
-    tick(300);
-    wizard.closable = true;
+  it('should be closed', () => {
```
Comment (msg-tbrtz) [inline projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:551]: naming: should be closed on what trigger?

**Review by msg-tbrtz — COMMENTED** [review]
overall, great refactoring!

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.ts (hunk 1 above)
Comment (msg-ptavoul) [reply projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.ts]: Done

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts (hunk 3 above)
Comment (msg-ptavoul) [reply projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:423]: Removed

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts (hunk 4 above)
Comment (msg-ptavoul) [reply projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:438]: Done

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts (hunk 5 above)
Comment (msg-ptavoul) [reply projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:444]: Done

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts (hunk 6 above)
Comment (msg-ptavoul) [reply projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:476]: Done

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts (hunk 7 above)
Comment (msg-ptavoul) [reply projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.spec.ts:551]: Done

**Review by msg-tbrtz — APPROVED** [review]

**Review by msg-tbrtz — APPROVED** [review]

File: projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.ts [hunk 8]
```diff
@@ -129,63 +130,75 @@ export class WizardComponent implements OnInit, AfterContentInit {
...
-  private resetStepper(): void {
-    this.stepper.reset();
-    this.stepperIndexChange.emit(0);
+  closeDialog(): void {
+    this.resetWizard();
+    this.close();
+  }
+
+  /**
+   * Is resetting the wizard position
```
Comment (msg-tbrtz) [inline projects/isy-angular-widgets/src/lib/wizard/components/wizard/wizard.component.ts:198]: *resets

**Review by msg-tbrtz — CHANGES_REQUESTED** [review]

**Review by msg-tbrtz — APPROVED** [review]

**Pull request merged**